ioctl/client_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package ioctl
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/pkg/util/fileutil"
+	"github.com/iotexproject/iotex-core/testutil"
+)
+
+func TestStop(t *testing.T) {
+	c := NewClient()
+	_, err := c.APIServiceClient(APIServiceConfig{Endpoint: "127.0.0.1:14014", Insecure: true})
+	require.NoError(t, err)
+	err = c.Stop(context.Background())
+	require.NoError(t, err)
+}
+
+func TestAskToConfirm(t *testing.T) {
+	c := NewClient()
+	defer c.Stop(context.Background())
+	blang := c.AskToConfirm()
+	// no input
+	require.False(t, blang)
+
+	c = &client{
+		cfg:  config.ReadConfig,
+		lang: config.Chinese,
+	}
+	blang = c.AskToConfirm()
+	require.False(t, blang)
+
+	c = &client{
+		cfg:  config.ReadConfig,
+		lang: 2, // other language is english as default
+	}
+	blang = c.AskToConfirm()
+	require.False(t, blang)
+}
+
+func TestAPIServiceClient(t *testing.T) {
+	c := NewClient()
+	defer c.Stop(context.Background())
+	apiServiceClient, err := c.APIServiceClient(APIServiceConfig{Endpoint: "127.0.0.1:14014", Insecure: true})
+	require.NoError(t, err)
+	require.NotNil(t, apiServiceClient)
+
+	apiServiceClient, err = c.APIServiceClient(APIServiceConfig{Endpoint: "127.0.0.199:14014", Insecure: false})
+	require.NoError(t, err)
+	require.NotNil(t, apiServiceClient)
+
+	apiServiceClient, err = c.APIServiceClient(APIServiceConfig{Endpoint: "", Insecure: false})
+	require.Error(t, err)
+	require.Contains(t, `use "ioctl config set endpoint" to config endpoint first`, err.Error())
+	require.Nil(t, apiServiceClient)
+}
+
+func TestGetAddress(t *testing.T) {
```

Comment:
There is a similar pattern in this func. Could it be written in a loop?

Reply:
yes, i use a func for middle process,  then the input and output may be clearly for many conditions, exp error messages.